packages/spatial/src/physics/components/ColliderComponent.tsx

```diff
@@ -23,20 +23,43 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
 Infinite Reality Engine. All Rights Reserved.
 */
 
-import { defineComponent, useComponent, useEntityContext, useOptionalComponent } from '@ir-engine/ecs'
-import { useState } from '@ir-engine/hyperflux'
+import {
+  defineComponent,
+  Entity,
+  EntityUUID,
+  hasComponent,
+  UndefinedEntity,
+  useComponent,
+  useEntityContext,
+  useOptionalComponent,
+  UUIDComponent
+} from '@ir-engine/ecs'
+import { defineState, getMutableState, none, useState } from '@ir-engine/hyperflux'
 
+import { ColliderDesc } from '@dimforge/rapier3d-compat'
 import { S } from '@ir-engine/ecs/src/schemas/JSONSchemas'
 import { useEffect, useLayoutEffect } from 'react'
 import { removeCallback, setCallback } from '../../common/CallbackComponent'
-import { useAncestorWithComponents } from '../../transform/components/EntityTree'
+import { MeshComponent } from '../../renderer/components/MeshComponent.ts'
+import {
+  getAncestorWithComponents,
+  getTreeFromChildToAncestor,
+  useAncestorWithComponents,
+  useChildrenWithComponents
+} from '../../transform/components/EntityTree'
 import { TransformComponent } from '../../transform/components/TransformComponent'
+import { computeTransformMatrix } from '../../transform/systems/TransformSystem.ts'
 import { Physics } from '../classes/Physics'
 import { CollisionGroups, DefaultCollisionMask } from '../enums/CollisionGroups'
-import { ShapeSchema, Shapes } from '../types/PhysicsTypes'
+import { Shapes, ShapeSchema } from '../types/PhysicsTypes'
 import { RigidBodyComponent } from './RigidBodyComponent'
 import { TriggerComponent } from './TriggerComponent'
 
+export const NestedCollidersState = defineState({
```

Comment:
a separate state like this is a bad idea. the collider component should contain all the data relevant for itself. this can be moved to the collider schema.

Reply:
removed